### ISSUE TITLE
az ml 2.24.0 not compatible with mcr.microsoft.com/azure-cli:2.58.0

### ISSUE THREAD

**Issue by danigian** [Azure/azure-cli-extensions#7385]
### Describe the bug

az ml 2.24.0 extension is not compatible with container image 'mcr.microsoft.com/azure-cli:2.58.0'

This is related to the newly added dependency on azureml-dataprep-rslex which has no musllinux compatible distributions

### Related command

az extension add --name ml --version 2.24.0

### Errors

az extension add --name ml --version 2.24.0

Default enabled including preview versions for extension installation now. Disabled in May 2024. Use '--allow-preview true' to enable it specifically if needed. Use '--allow-preview false' to install stable version only. 
An error occurred. Pip failed with status code 1. Use --debug for more information.

### Issue script & Debug output

```
...
INFO: pip is looking at multiple versions of ml to determine which version is compatible with other requirements. This could take a while.
Will try a different candidate, due to conflict:
    ml 2.24.0 depends on azureml-dataprep-rslex>=2.22.0
ERROR: Could not find a version that satisfies the requirement azureml-dataprep-rslex>=2.22.0 (from ml) (from versions: none)
ERROR: No matching distribution found for azureml-dataprep-rslex>=2.22.0
...
```

### Expected behavior

I should be able to install the ML extension on a Microsoft-provided Azure CLI container image

### Environment Summary

azure-cli                         2.58.0

core                              2.58.0
telemetry                          1.1.0

Dependencies:
msal                              1.26.0
azure-mgmt-resource             23.1.0b2

Python location '/usr/local/bin/python'
Extensions directory '/root/.azure/cliextensions'

Python (Linux) 3.11.8 (main, Feb  7 2024, 23:20:20) [GCC 13.2.1 20231014]

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by jsburckhardt** [comment]
same here - I reproduced it within the container and try install ml extension. Meanwhile if you install v2.23.0 it works

```bash
az extension add --name ml --version  2.23.0
```

more details about the error:
![image](https://github.com/Azure/azure-cli-extensions/assets/18494471/c718cee0-1d42-4532-a491-4fb399b01714)

**Comment by YusakuNo1** [comment]
**Summary:** 
1. CLI 2.24.0 introduces a hard dependency to package `azureml-dataprep-rslex`.  `azureml-dataprep-rslex` was the dependency for ML extension in 3+ years, but it was optional.
2.  `azureml-dataprep-rslex` doesn't have a release for Linux Alpine.  `azureml-dataprep-rslex` team claimed to support Linux Alpine could be hard and they don't have requests to support Linux Alpine distribution.

**Options:**
* Remove the hard dependency of `azureml-dataprep-rslex` from Azure CLI ML extension. With this fix, Linux Alpine users can install Azure CLI ML extension, but all the users can't user az ml data mount, when they are using, we will show a message "please install azureml-dataprep-rslex".
* (Need some help from Azure CLI team) Skip the metadata checking for this dependency. With this fix, Linux Alpine users can install Azure CLI ML extension, and only the users in Linux Alpine distribution will get affected by the error message from #1 above.

**Comment by YusakuNo1** [comment]
FYI: the fix is released at CLI 2.25.1, please verify

**Comment by YusakuNo1** [comment]
Close the issue as it's fixed
